tests/components/sensor/test_hddtemp.py

```diff
@@ -0,0 +1,133 @@
+"""The tests for the hddtemp platform."""
+import unittest
+from unittest.mock import patch
+
+from homeassistant.setup import setup_component
+
+from tests.common import (get_test_home_assistant, load_fixture)
+
+VALID_CONFIG_MINIMAL = {
+    'sensor': {
+        'platform': 'hddtemp',
+    }
+}
+
+VALID_CONFIG_NAME = {
+    'sensor': {
+        'platform': 'hddtemp',
+        'name': 'FooBar',
+    }
+}
+
+VALID_CONFIG_ONE_DISK = {
+    'sensor': {
+        'platform': 'hddtemp',
+        'disks': [
+            '/dev/sdb1',
+        ],
+    }
+}
+
+VALID_CONFIG_MULTIPLE_DISKS = {
+    'sensor': {
+        'platform': 'hddtemp',
+        'host': 'foobar.local',
+        'disks': [
+            '/dev/sda1',
+            '/dev/sdb1',
+            '/dev/sdc1',
+        ],
+    }
+}
+
+
+class TelnetMock():
+    """Mock class for the telnetlib.Telnet object."""
+
+    def __init__(self, host, port, timeout=0):
+        """Initialize Telnet object."""
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+        self.sample_data = bytes(load_fixture('hddtemp.txt'), 'ascii')

```

Comment:
The fixture is pretty small. Is there an advantage beside the cleaner code of not including it here and skip the loading from file?